src/app/core/user/user-account/user-account.component.ts

```diff
@@ -33,38 +34,33 @@ export class UserAccountComponent implements OnInit {
   /** user to be edited */
   username: string;
 
-  /** whether password change is disallowed because of demo mode */
-  disabledForDemoMode: boolean;
-  disabledForOfflineMode: boolean;
+  passwordChangeDisabled = false;
+  tooltipText;
+  couchdbAuthService: CouchdbAuthService;
 
   constructor(
     private sessionService: SessionService,
     public authService: AuthService
-  ) {}
+  ) {
+    if (this.authService instanceof CouchdbAuthService) {
+      this.couchdbAuthService = this.authService;
+    }
```

Comment:
wouldn't it be more straight forward to let the password component inject AuthService itself through DI and then simply disable itself in case it's not the right type, instead of leaking all this into the user account component?